src/components/FlowChart/Sidebar.jsx

```diff
@@ -333,8 +333,17 @@ class FlowChartSidebar extends React.Component {
     );
   };
 
+  renderReadOnly = () =>{
+    return(<div className="readonly-overlay py-4 px-4 text-center">
+      <h4 className="my-4"><b>This flow is <span className="text-muted">readonly</span></b></h4>
+      <p className="my-4">To edit this flow, please duplicate it.</p>
+      <Button onClick={this.props.duplicateFlow}>Duplicate Flow</Button>
+    </div>)
+  }
+
```

Comment:
Why `renderReadOnly`?
If it's big enough for this method, it should be a component. If it's too small, it should be inlined.

Reply:
This follows the structure of the rest of the sidebar, I will refactor the sidebar to separate the renders into components